contracts/pact_state_handlers_methods.go

```diff
@@ -0,0 +1,170 @@
+		createdHasComp := &appstudiov1alpha1.Component{}
+
+		// wait until component is created
+		gomega.Eventually(func() bool {
+			err := k8sClient.Get(context.Background(), hasCompLookupKey, createdHasComp)
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+			return len(createdHasComp.Status.Conditions) > 1
+		}, timeout, interval).Should(gomega.BeTrue())
+
+		// wait until component is ready
+		gomega.Eventually(func() bool {
+			err := k8sClient.Get(context.Background(), hasAppLookupKey, createdHasApp)
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+			return len(createdHasApp.Status.Conditions) > 0 && strings.Contains(createdHasApp.Status.Devfile, comp.name)
+		}, timeout, interval).Should(gomega.BeTrue())
+	}
+	return nil, nil
+
+}
+
+func removeAllInstacesInNamespace(namespace string, myInstance client.Object) {
+	objectKind := strings.Split(reflect.TypeOf(myInstance).String(), ".")[1]
+	remainingCount := getObjectCountInNamespace(objectKind, namespace)
+	if remainingCount == 0 {
+		return
+	}
+	// remove resources in namespace
+	err := k8sClient.DeleteAllOf(context.Background(), myInstance, client.InNamespace(namespace))
+	gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+	// watch number of resources existing
+	gomega.Eventually(func() bool {
+		objectKind := strings.Split(reflect.TypeOf(myInstance).String(), ".")[1]
+		remainingCount := getObjectCountInNamespace(objectKind, namespace)
+		println("Removing", objectKind, "instances from", namespace, "namespace. Remaining: ", remainingCount)
+		return remainingCount == 0
+	}, timeout, interval).Should(gomega.BeTrue())
+}
+
+func getObjectCountInNamespace(objectKind string, namespace string) int {
+	unstructuredObject := &unstructured.Unstructured{}
+
+	unstructuredObject.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   appstudiov1alpha1.GroupVersion.Group,
+		Version: appstudiov1alpha1.GroupVersion.Version,
+		Kind:    objectKind,
+	})
+
+	err := k8sClient.List(context.Background(), unstructuredObject, &client.ListOptions{Namespace: namespace})
+	gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+	listOfObjects, _ := unstructuredObject.ToList()
+	return len(listOfObjects.Items)
+}
+
+func cleanUpNamespaces() {
+	println("clean up namespaces")
+	removeAllInstances(&appstudiov1alpha1.Component{})
+	removeAllInstances(&appstudiov1alpha1.Application{})
+	removeAllInstances(&appstudiov1alpha1.ComponentDetectionQuery{})
+}
+
+// remove all instances of the given type within the whole cluster
+func removeAllInstances(myInstance client.Object) {
+	listOfNamespaces := getListOfNamespaces()
+	for _, item := range listOfNamespaces.Items {
+		removeAllInstacesInNamespace(item.Name, myInstance)
+	}
+}
+
+// return all namespaces where the instances of the specified object kind exist
+func getListOfNamespaces() core.NamespaceList {
+	namespaceList := &core.NamespaceList{}
+	err := k8sClient.List(context.Background(), namespaceList, &client.ListOptions{Namespace: ""})
```

Comment:
You are not printing the error in gomega expect. In case of debug 0 information give this block